editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5213,6 +5213,7 @@ CanvasItemEditor::CanvasItemEditor(EditorNode *p_editor) {
 
 	zoom_widget = memnew(EditorZoomWidget);
 	controls_vb->add_child(zoom_widget);
+	controls_vb->move_child(zoom_widget, 0);
```

Comment:
That's a bit surprising, what children are present in `controls_vb` before this one? `controls_vb` is created on line 5186 and I don't see any other `controls_vb->add_child()` call before this one.

Reply:
Other child nodes are added in CanvasItemEditorViewport constructor, which is created before this line. They use `get_controls_container()->add_child()`.

Maybe instead of using `move_child()`, it would be better to move the zoom widget creation before that? It could happen right after `controls_vb` is created.

Reply:
Thanks, that makes sense. Then yeah this code should be moved to right after:
```
	controls_vb = memnew(VBoxContainer);
	controls_vb->set_begin(Point2(5, 5));
```
And the `move_child` wouldn't be needed.

Reply:
I've moved the `zoom_widget` constructor to immediately after the `controls_vb` constructor and removed the call to `move_child`